tests/rptest/tests/partition_force_reconfiguration_test.py

```diff
@@ -502,6 +582,21 @@ def no_pending_force_reconfigurations():
                     self.redpanda._admin.await_stable_leader(
                         topic=topic.name,
                         partition=part,
-                        timeout_s=self.WAIT_TIMEOUT_S,
+                        timeout_s=self.default_timeout_sec,
                         backoff_s=2,
                         hosts=self._alive_nodes())
+        topic_hws_after_recovery = {}
+        for t in topics:
+            topic_hws_after_recovery[
+                t.name] = self.get_topic_partition_high_watermarks(t.name)
+
+        for t in topics:
+            for id, (initial_hw, final_hw) in enumerate(
```

Comment:
Regaining partition id from the list index looks quite cheeky to me, although I cannot see how it may get broken if we don't delete partitions.